client/core/trade.go

```diff
@@ -2324,24 +2390,24 @@ func (c *Core) refundMatches(t *trackedTrade, matches []*matchTracker) (uint64,
 			swapCoinID = dex.Bytes(match.MetaData.Proof.TakerSwap)
 			matchFailureReason = "no valid redemption received from Maker"
 		default:
-			t.dc.log.Errorf("attempted to execute invalid refund for match %s, side %s, status %s",
+			c.log.Errorf("attempted to execute invalid refund for match %s, side %s, status %s",
 				match, match.Side, match.Status)
 			continue
 		}
 
 		swapCoinString := coinIDString(refundAsset.ID, swapCoinID)
-		t.dc.log.Infof("Refunding %s contract %s for match %s (%s)",
+		c.log.Infof("Refunding %s contract %s for match %s (%s)",
 			refundAsset.Symbol, swapCoinString, match, matchFailureReason)
 
-		var feeSuggestion uint64
+		var feeRate uint64
 		if _, is := t.accountRefunder(); is {
-			feeSuggestion = t.metaData.MaxFeeRate
+			feeRate = t.metaData.MaxFeeRate
 		}
-		if feeSuggestion == 0 {
-			feeSuggestion = c.feeSuggestionAny(refundAsset.ID)
+		if feeRate == 0 {
+			feeRate = c.feeSuggestionAny(refundAsset.ID) // includes wallet itself
 		}
 
-		refundCoin, err := refundWallet.Refund(swapCoinID, contractToRefund, feeSuggestion)
+		refundCoin, err := refundWallet.Refund(swapCoinID, contractToRefund, feeRate)
```

Comment:
We're already getting a proper fee rate for Refund, so we can just solidify the new semantics and avoid a redundant internal fee rate request.